website/content/docs/v0.9/Guides/converting-control-plane.md

```diff
@@ -0,0 +1,247 @@
+talos-default-master-1   Ready    control-plane,master   58m   v1.20.4   172.20.0.2    <none>        Talos (v0.9.0)   5.10.19-talos    containerd://1.4.4
+talos-default-master-2   Ready    control-plane,master   58m   v1.20.4   172.20.0.3    <none>        Talos (v0.9.0)   5.10.19-talos    containerd://1.4.4
+talos-default-master-3   Ready    control-plane,master   58m   v1.20.4   172.20.0.4    <none>        Talos (v0.9.0)   5.10.19-talos    containerd://1.4.4
+talos-default-worker-1   Ready    <none>                 58m   v1.20.4   172.20.0.5    <none>        Talos (v0.9.0)   5.10.19-talos    containerd://1.4.4
+```
+
+Start the conversion script:
+
+```bash
+$ talosctl -n <IP> convert-k8s
+discovered master nodes ["172.20.0.2" "172.20.0.3" "172.20.0.4"]
+current self-hosted status: true
+gathering control plane configuration
+aggregator CA key can't be recovered from bootkube-boostrapped control plane, generating new CA
+patching master node "172.20.0.2" configuration
+patching master node "172.20.0.3" configuration
+patching master node "172.20.0.4" configuration
+waiting for static pod definitions to be generated
+waiting for manifests to be generated
+Talos generated control plane static pod definitions and bootstrap manifests, please verify them with commands:
+    talosctl -n <master node IP> get StaticPods.kubernetes.talos.dev
+    talosctl -n <master node IP> get Manifests.kubernetes.talos.dev
+
+bootstrap manifests will only be applied for missing resources, existing resources will not be updated
+confirm disabling pod-checkpointer to proceed with control plane update [yes/no]:
+```
+
+Script stops at this point waiting for confirmation.
+Talos still runs self-hosted control plane, and static pods were not rendered yet.
+
+As instructed by the script, please verify that static pod definitions are correct:
+
+```bash
+$ talosctl -n <IP> get staticpods -o yaml
+node: 172.20.0.2
+metadata:
+    namespace: controlplane
+    type: StaticPods.kubernetes.talos.dev
+    id: kube-apiserver
+    version: 1
+    phase: running
+spec:
+    apiVersion: v1
+    kind: Pod
+    metadata:
+        annotations:
+            talos.dev/config-version: "2"
+            talos.dev/secrets-version: "1"
+        creationTimestamp: null
+        labels:
+            k8s-app: kube-apiserver
+            tier: control-plane
+        name: kube-apiserver
+        namespace: kube-system
+    spec:
+        containers:
+            - command:
+...
+```
+
+Static pod definitions are generated from machine configuration and should match pod template as generated by Talos on bootstrap of self-hosted control plane unless there were some manual changes applied to the daemonset specs after bootstrap.
+Talos patches machine configuration with container image versions scraped from the daemonset definition, fetches service account key from Kubernetes secrets.
+Aggregator CA can't be recovered from the self-hosted control plane, so new CA gets generated.
+
```

Comment:
An additional note should be added here to tell the user to say `NO` to this for the first run, so that they can verify the manifests, as indicated below.

Reply:
technically they don't have to say `no`, as they can use another window to perform lookups.

Reply:
My key point here is that we want to prevent people from blindly saying "yes" before they have inspected things.  The pipelining is convenient, but if it requires branching to perform checks, we should make that apparent _before_ they blindly say "yes".